python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1958,7 +1959,7 @@ def get_latest_tags_by_partition(
 
     def get_latest_asset_partition_materialization_attempts_without_materializations(
```

Comment:
Would it make sense to rename this to `get_latest_asset_partition_records`? It seems like "Fetch the latest materialzation and materialization planned events for each partition" isn't exactly accurate anymore?

Reply:
I haven't changed anything about the criteria we are using to decide which rows to return - the method still only returns records that correspond to planned events that don't have a corresponding materialization event, it just now returns more information about each of those rows

(This is kind of an odd method - it is a bit tailormade for the specific information that the asset status partition cache)

Reply:
i'm fine to change the name, but the "attempts_without_materializations" aspect is still relevant

Reply:
Oy yeah a lot going on here. I think maybe this would do it?

```
# current:
get_latest_asset_partition_materialization_attempts_without_materializations

# accurate in light of new return type:
get_latest_asset_partitions_with_materialization_attempts_without_materializations
```

This line in the docstring probably needs to change too: "Returns a mapping of partition to [run id, event id]."

Reply:
or `get_latest_asset_partition_records_with_incomplete_materializations`?

Reply:
i think that last one probably is OK if we're ok with materialization failures being considered "incomplete" (although they did complete, just unsuccessfully...)

Reply:
if we're set on adding AssetPartitionRecords, my preference would be to create a whole new storage method that supports pagination / cursoring, with filters for successful/incomplete materializations